core/src/blocktree.rs

```diff
@@ -1305,6 +1309,61 @@ impl Blocktree {
         self.db.write(batch)?;
         Ok(())
     }
+
+    fn should_insert_blob(
+        &self,
+        slot: &SlotMeta,
+        prev_inserted_blob_datas: &HashMap<(u64, u64), &[u8]>,
+        blob: &Blob,
+    ) -> bool {
+        let blob_index = blob.index();
+        let blob_slot = blob.slot();
+
+        // Check that the blob doesn't already exist
+        if blob_index < slot.consumed
+            || prev_inserted_blob_datas.contains_key(&(blob_slot, blob_index))
+            || self.data_blob_exists(blob_slot, blob_index)
```

Comment:
@mark-solana, does this look like the correct way to check for blob existence?

Reply:
I think that check is not useful, because the metadata would be stale. Also might give incorrect results if a blob has been deleted.  You'd have to check the metadata in the current `erasure_meta_working_set`. 

But I think that's also not useful, because if that has been updated to indicate that the blob is present, then that blob should already be in `prev_inserted_blob_datas`.